source/__tests__/integration/export-profile.test.js

```diff
@@ -1,33 +1,57 @@
-const path = require('path')
-
-describe('Sinopia Profile Editor exports a loaded Profile', () => {
+describe('Sinopia Profile Editor exports an edited profile', () => {
 
   beforeAll(async () => {
-    await page.goto('http://127.0.0.1:8000/#/profile/create/')
+    await page.goto('http://localhost:8000/#/profile/create/true')
   })
 
-  it('error displayed when exporting profile form data', async () => {
-     await expect(page).toClick('a', { text: 'Export'})
-     const alert_box_header = await page.$eval('#alertBox > div > div > div.modal-header > h3', e => e.textContent)
-     await expect(alert_box_header).toMatch(/Error!/)
- })
-
-  describe('Exporting an existing profile', () => {
+  describe('Exporting an edited profile', () => {
 
     beforeAll(async () => {
-      // await expect(page).toClick('a', { text: 'Import'})
+      const path = require('path')
       const bf_item_location = path.join(__dirname, "..", "__fixtures__", 'item.json')
       await expect(page).toUploadFile(
-        'input[type="file"]',
-        bf_item_location,
+          'input[type="file"]',
+          bf_item_location,
       )
     })
 
-    it('saves a non-empty Profile', async () => {
+    it('imports an existing profile', async () => {
+      await page.waitForSelector('span[popover-title="Profile ID: profile:bf2:Item"]') // Waits for the Profile to load
+    })
+
+    it('changes some values', async () => {
+      await page.waitFor(1000)
+      await page.$eval('form[name="profileForm"]', e => e.reset())
+      await expect(page).toFillForm('form[name="profileForm"]', {
+        id: "profile:my:Item",
+        description: "my item profile",
+        author: "Me",
+        title: "My Profile"
+      })
+    }, 10000)
+
+    it('exports JSON data for a non-empty Profile', async () => {
+      let data;
+
+      page.click('a.import-export')
+          .then(async () => {
+            console.log('Export button clicked')
+          })
+          .catch(e => console.log(`failed promise on clicking the Export button: ${e}`))
+
       await page.waitFor(1000)
-      await expect(page).toClick('a', { text: 'Export'})
-      // TODO: test that profile written to file system (see #77)
-      // TODO: test fields that are validated, listed in source/assets/js/modules/profile/services/profileHandler.service.js
+
+      page.waitForSelector('a[download="My Profile.json"]')
+          .then(
+            data = await page.$eval('a[download="My Profile.json"]', e => e.getAttribute('href'))
+          )
+          .catch(e => console.log(`failed promise on download data link: ${e}`))
+
+      const json = JSON.parse(data.substr(data.indexOf(',') + 1))
+      expect(json['Profile']['id']).toBe('profile:my:Item')
+      expect(json['Profile']['description']).toBe('my item profile')
+      expect(json['Profile']['author']).toBe('Me')
+      expect(json['Profile']['title']).toBe('My Profile')
```

Comment:
Yes. in the `export.directive.js:32` file, it ends up doing a click() on that data link in order to download it. Otherwise all of the data is there in the link.